session-manager/src/session_manager/session_manager_base.py

```diff
@@ -0,0 +1,336 @@
+import json
+import logging
+import uuid
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+class SessionManagerBase(ABC):
+    """
+    Base class for session managers with common functionality.
+    """
+
+    ROUTING_AGENT_NAME = "routing-agent"
+
+    def __init__(self, agent_manager):
+        """
+        Initializes the SessionManagerBase.
+
+        Args:
+            agent_manager: An agent manager instance (AgentManager or ResponsesAgentManager).
+        """
+        self.agent_manager = agent_manager
+        self.agents = agent_manager.agents()
+        self.user_sessions = {}
+
+        # Optional session metadata persistence
+        self.metadata_file = None  # Subclasses can set this to enable persistence
+        self.session_metadata = {}
+
+    def _generate_session_name(self, user_id: str, agent_name: str = None) -> str:
+        """Generate a unique session name"""
+        unique_id = str(uuid.uuid4())[:8]  # First 8 characters of UUID
```

Comment:
is it possible to have more sessions from the same user?

Reply:
This would depend on the integration. For the command like cli, there can be several concurrent sessions and they can each be the same user. For slack integration I think we can only have one session at a time for a given user as all chats from that user will be mapped to the existing session if it exists.